src/pages/EditProduct.js

```diff
@@ -1,7 +1,64 @@
 import React from 'react';
+import { useParams } from 'react-router-dom';
+import { updateProduct } from './api'
 
-const EditProduct = () => {
-  return <div>Edit a product</div>;
+const EditProduct = ({ token, products, navigate }) => {
+    const { productId } = useParams();
+    const [currentProduct] = products.filter(product => product.id === parseInt(productId));
+
+    if (currentProduct === undefined) {
+        return null;
+    }
+
+    const { name, description, price, thumbnailImage } = currentProduct;
+
+    const [newName, setNewName] = useState(name);
+    const [newDescription, setNewDescription] = useState(description);
+    const [newPrice, setNewPrice] = useState(price);
+    const [newThumbnailImage, setNewThumbnailImage] = useState('');
+
+    async function editProduct() {
+        const updatedProduct = {
+            token: token,
+            name: newName,
+            description: newDescription,
+            price: newPrice,
+            thumbnailImage: newThumbnailImage,
+            productId
+        }
+        const results = await updateProduct(updatedProduct)
+    }
+
+    return (
+        <div className="editProductContainer">
+            <form onSubmit={(event) => {
```

Comment:
I would recommend using <Form> component from Bootstrap library, comes with great out of the box formatting and additional functionality